File: responser-chrome-extension/src/app/model/Review.ts
import {User} from "./User";
import moment, {Moment} from "moment";
import {ReviewLike} from "./ReviewLike";
import {ResourceType} from "./ResourceType";
import {WebResource} from "./WebResource";

export class Review {

    id: string;

    resourceId: string;

    reviewId: string;

    rating: number;

    text: string;

    user: User;

    reviewLikes: ReviewLike[];

    webResource: WebResource;

    creationDate: Moment;

    updateDate: Moment;

    public static deserialize(data: any): Review {
        return Object.assign(new Review(), data, {
            user: User.deserialize(data.user),
            creationDate: moment(data.creationDate),
            updateDate: moment(data.updateDate),
            reviewLikes: data.reviewLikes.map((like: any) => ReviewLike.deserialize(like)),
            webResource: data.webResource ? WebResource.deserialize(data.webResource) : null
        });
    }
}

File: reviewly-backend/src/main/java/space/reviewly/backend/model/Review.java
package space.reviewly.backend.model;

import jakarta.persistence.*;
import lombok.*;

import java.util.Set;
import space.reviewly.backend.model.user.User;

/**
 * Review
 *
 * @author Shcherbachenya Igor
 */
@Getter
@Setter
@RequiredArgsConstructor
@Entity
@Table(name = "reviews")
public class Review extends AbstractEntity {

    @Column(name = "user_id")
    private String userId;

    @Column(name = "resource_id")
    private String resourceId;

    @Column(name = "review_id")
    private String reviewId;

    private Byte rating;

    private String text;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "resource_id", nullable = false, insertable = false, updatable = false)
    private WebResource webResource;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false, insertable = false, updatable = false)
    private User user;

    @OneToMany(mappedBy = "review", fetch = FetchType.LAZY)
    private Set<ReviewLike> likes;
}
